File: 백준/Silver/20922. 겹치는 건 싫어/겹치는 건 싫어.py
import sys
read = sys.stdin.readline

n, k = map(int, read().split())
seq = list(map(int, read().split()))
board = [0] * 100001

l, r = 0, 0
ans = 1

while r < n:
    if l == r:
        board[seq[r]] += 1
        r += 1
        continue

    if board[seq[r]] == k:
        ans = max(ans, r-l)
        while board[seq[r]] == k:
            board[seq[l]] -= 1
            l += 1
    else:
        board[seq[r]] += 1
        r += 1
print(max(ans, r - l))

File: 백준/Silver/20922. 겹치는 건 싫어/겹치는 건 싫어.js
const input = require('fs').readFileSync(process.platform === "linux" ? "/dev/stdin" : "./sample.txt").toString().trim().split("\n");

const [n, m] = input[0].split(" ").map(Number);
const seq = input[1].split(" ").map(Number);

const store = new Map();
let ans = 1;
let [l, r] = [0, 1];
store.set(seq[l], 1);

while (l <= r) {
  if (r === n) {
    ans = Math.max(ans, r - l);
    break;
  } else {
    let cnt = 1;
    if (store.has(seq[r])) {
      cnt = store.get(seq[r]) + 1;
    }

    if (cnt > m) {
      ans = Math.max(ans, r - l);
      while (store.get(seq[r]) === m) {
        let tmp = store.get(seq[l]);
        store.set(seq[l], tmp - 1);
        l += 1;
      }
      store.set(seq[r], cnt - 1);
      r += 1;
    } else {
      store.set(seq[r], cnt);
      r += 1;
    }
  }
}
console.log(ans);